treeherder/model/models.py

```diff
@@ -1477,11 +1525,69 @@ class Meta:
         db_table = "text_log_error"
         unique_together = ('step', 'line_number')
 
+    def __str__(self):
+        return "{0} {1}".format(self.id, self.step.job.id)
+
+    @property
+    def metadata(self):
+        try:
+            return self._metadata
+        except TextLogErrorMetadata.DoesNotExist:
+            return None
+
     def bug_suggestions(self):
         from treeherder.model import error_summary
         return error_summary.bug_suggestions_line(self)
 
 
+class TextLogErrorMetadata(models.Model):
+    class Meta:
+        db_table = "text_log_error_metadata"
```

Comment:
To be consistent with other models in this file, let's put this after the property declarations.